pyface/font.py

```diff
@@ -0,0 +1,314 @@
+
+Font Family
+    A list of font family names in order of preference, such as "Helvetica"
+    or "Comic Sans".  There are several generic font family names that can
+    be used as fall-backs in case all preferred fonts are unavailable.  In
+    the case of a font that has been selected by the toolkit this list will
+    have one value which is the actual font family name.
+
+Weight
+    How thick or dark the font glyphs are.  These can be given as a number
+    from 1 (lightest) to 999 (darkest), but are typically specified by a
+    multiple of 100 from 100 to 900, with a number of synonyms such as 'light'
+    and 'bold' available for those values.
+
+Stretch
+    The amount of horizontal compression or expansion to apply to the glyphs.
+    These can be given as a percentage between 50% and 200%, or by strings
+    such as as 'condensed' and 'expanded' that correspond to those values.
+
+Style
+    This selects either 'oblique' or 'italic' variants typefaces of the given
+    font family.  If neither is wanted, the value is 'normal'.
+
+Size
+    The overall size of the glyphs. This can be expressed either as the
+    numeric size in points, or as a string such as "small" or "large".
+
+Variants
+    A set of additional font style specifiers, such as "small-caps",
+    "strikethrough", "underline" or "overline", where supported by the
+    underlying toolkit.
+
+Font Specificiation Class
+-------------------------
+
+The Pyface Font class is a HasStrictTraits class which specifies a requested
+font.  It has traits for all of the font properties, plus additional utility
+methods that produce modifed versions of the font.
+
+It also has methods that convert the Font class to and from a toolkit Font
+class.
+
+"""
+import re
+
+from traits.api import (
+    BaseCFloat, CList, CSet, DefaultValue, Enum, HasStrictTraits, Map, Range,
+    Str, TraitError, TraitType
+)
+from traits.trait_type import NoDefaultSpecified
+
+WEIGHTS = {str(i): i for i in range(100, 1001, 100)}
+
+# Note: we don't support 'medium' as an alias for weight 500 because it
+# conflicts with the usage of 'medium' as an alias for a 12pt font in the CSS
+# specification for font attributes.
+WEIGHTS.update({
+    'thin': 100,
+    'extra-light': 200,
+    'ultra-light': 200,
+    'light': 300,
+    'normal': 400,
+    'regular': 400,
+    'book': 400,
+    'semibold': 600,
+    'demibold': 600,
+    'demi': 600,
+    'bold': 700,
+    'extra-bold': 800,
+    'ultra-bold': 800,
+    'black': 900,
+    'heavy': 900,
+    'extra-heavy': 1000,
+})
+
+STRETCHES = {
+    'ultra-condensed': 50,
+    'ultracondensed': 62.5,
+    'extra-condensed': 62.5,
+    'extracondensed': 62.5,
+    'condensed': 75,
+    'semi-condensed': 87.5,
+    'semicondensed': 87.5,
+    'normal': 100,
+    'semi-expanded': 112.5,
+    'semiexpanded': 112.5,
+    'expanded': 125,
+    'extra-expanded': 150,
+    'extraexpanded': 150,
+    'ultra-expanded': 200,
+    'ultraexpanded': 200,
+}
```

Comment:
Same here, CSS has these values: `normal | ultra-condensed | extra-condensed | condensed | semi-condensed | semi-expanded | expanded | extra-expanded | ultra-expanded`

I don't think we need the extra ones here if we document the fact that these values follow CSS convention.